### PR TITLE
CEO PR# 5.3 - CEOs Apollo, Duncan, Will

### PR DESCRIPTION
**Pull request by d-little**
### Apollo

Pretty straight forward

### Duncan

Needed some changes to Player.ts and uncommenting code in CeoExtension.ts

#### Player.ts

I _could_ have done this entirely in CeoExtension.ts, but am pretty sure that player.getCeo() will come in handy in the future.

### Will

Another easy one.

### CODE REVIEW

**Comment by d-little** [comment]
Im actually not 100% sure that the work I did with Duncan is right, get `getCeo()` isnt.. perfect.   I might pull out the `getCeo()` and put it all into the `CeoExtension.ts`.  Maybe if we do need it in the future it can be added?

File: src/server/Player.ts [hunk 1]
```diff
@@ -207,6 +208,10 @@ export class Player {
     return this.corporations.find((c) => c.name === corporationName);
   }
 
+  public getCeo(ceoName: CardName): ICeoCard | undefined {
```
Comment (kberg) [inline src/server/Player.ts:211]: How does this compile? It couldn't because the playedCards are of type Array<IProjectCard> or something.

**Comment by d-little** [comment]
> Im actually not 100% sure that the work I did with Duncan is right, get `getCeo()` isnt.. perfect. I might pull out the `getCeo()` and put it all into the `CeoExtension.ts`. Maybe if we do need it in the future it can be added?


 #5296 uses `getCeo()` for Xavier, so it's likely needed

File: src/server/cards/ceos/Will.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return this.isDisabled === false;
+  }
+
+  public action(player: Player): PlayerInput | undefined {
+    player.game.defer(new AddResourcesToCard(player, CardResource.ANIMAL, {count: 2}));
+    player.game.defer(new AddResourcesToCard(player, CardResource.MICROBE, {count: 2}));
+    player.game.defer(new AddResourcesToCard(player, CardResource.SCIENCE, {count: 1}));
+    player.game.defer(new AddResourcesToCard(player, CardResource.FLOATER, {count: 1}));
+    player.game.defer(new AddResourcesToCard(player, CardResource.ASTEROID, {count: 1}));
+    player.game.defer(new AddResourcesToCard(player, undefined, {count: 1}));
```
Comment (kberg) [inline src/server/cards/ceos/Will.ts:46]: Should be fine, but I'd like to confirm how this will work with Pathfinders' communication center.

Comment (d-little) [reply src/server/cards/ceos/Will.ts:46]: Added a check for this

File: src/server/Player.ts (hunk 1 above)
Comment (d-little) [reply src/server/Player.ts:211]: ¯\\_(ツ)_/¯

**Comment by d-little** [comment]
This is on pause till #5298 is in merged.  Once that's in I'll fix up the cards in this PR to match

File: src/server/cards/ceos/CeoCardManifest.ts [hunk 3]
```diff
@@ -32,10 +32,10 @@ import {Naomi} from './Naomi';
 // import {Tate} from './Tate';
 import {Ulrich} from './Ulrich';
 // import {VanAllen} from './VanAllen';
-import {Will} from './Will';
-// import {Xavier} from './Xavier';
-// import {Yvonne} from './Yvonne';
-// import {Zan} from './Zan';
+// import {Will} from './Will';
```
Comment (kberg) [inline src/server/cards/ceos/CeoCardManifest.ts:35]: Intentional?

File: src/server/cards/ceos/CeoCardManifest.ts [hunk 4]
```diff
@@ -72,9 +72,9 @@ export const CEO_CARD_MANIFEST = new ModuleManifest({
     // [CardName.TATE]: {Factory: Tate},
     [CardName.ULRICH]: {Factory: Ulrich},
     // [CardName.VANALLEN]: {Factory: VanAllen},
-    [CardName.WILL]: {Factory: Will, compatibility: 'venus'},
-    // [CardName.XAVIER]: {Factory: Xavier, compatibility: 'prelude'},
-    // [CardName.YVONNE]: {Factory: Yvonne, compatibility: 'colonies'},
-    // [CardName.ZAN]: {Factory: Zan, compatibility: 'turmoil'},
+    // [CardName.WILL]: {Factory: Will, compatibility: 'venus'},
```
Comment (kberg) [inline src/server/cards/ceos/CeoCardManifest.ts:75]: Same

**Review by kberg — APPROVED** [review]

**Pull request merged**